addons/mail/populate/mail_channel.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+from odoo.tools import populate
+
+
+class Channel(models.Model):
+    _inherit = "mail.channel"
+    _populate_dependencies = ["mail.message", "res.partner"]
```

Comment:
This dependency order is strange, I would be expecting messages to be created after channels and not before.